source/adapters/level_zero/command_buffer.cpp

```diff
@@ -437,6 +437,21 @@ urCommandBufferReleaseExp(ur_exp_command_buffer_handle_t CommandBuffer) {
 
 UR_APIEXPORT ur_result_t UR_APICALL
 urCommandBufferFinalizeExp(ur_exp_command_buffer_handle_t CommandBuffer) {
+  // Reset the L0 events we use for command-buffer internal sync-points to the
+  // non-signalled state
+  ZE2UR_CALL(zeCommandListAppendBarrier,
+             (CommandBuffer->ZeCommandList, nullptr, 0, nullptr));
```

Comment:
This will sync the entire queue, right? Is this the intention? Why not just wait for the specific events you are trying to reset here?

Reply:
Good point, yes this would wait on the whole queue, which is unnecessary. Updated to wait on the specific events. I've also set the signal event parameter of the barrier to signal an event that we we're previously using a standalone `zeCommandListAppendSignalEvent` command for.